libs/blocks/columns/columns.js

```diff
@@ -1,19 +1,86 @@
+const getMeaningfulChildren = (col) => Array.from(col.childNodes).filter(
+  (node) => node.nodeType !== Node.TEXT_NODE || node.textContent.trim() !== '',
+);
+
+const isRowHeader = (cols, col, cdx, rows) => {
+  const columnsWithStrongOrH = Array.from(cols)
+    .filter((rowCol) => getMeaningfulChildren(rowCol)
+      .some((child) => child.tagName?.match(/^(H[1-6]|STRONG)$/)));
+
+  const firstRow = rows[0];
+  const firstRowCols = firstRow.querySelectorAll(':scope > div');
```

Comment:
NIT: If `firstRow` is only used here, there is no need for variable
```suggestion
  const firstRowCols = row[0].querySelectorAll(':scope > div');
```